### PR TITLE
Fix/exclude

### LINKED ISSUE #6: The `!` filter doesnt work
I would like to do something like this: `['*', '!name']` - allow all fields but exclude `name` from it. It seems to be not working.

### PR DESCRIPTION
**Pull request by stefanprokopdev**
I want to resolve #6. What is new:

- Node version
- Fixed audit
- Exclude function - almost same as `filter` func, **do you have better idea / solution?** Maybe I can refactor the same parts of code in these functions
- Integration tests

I cant find a way how to exclude fields via `jsonpath`

### CODE REVIEW

File: src/test/acl.test.ts [hunk 1]
```diff
@@ -196,7 +196,6 @@ describe('ACL', () => {
         const allowedBook = permission.filter(book);
 
         expect(permission.granted).toBe(true);
-        expect(permission.attributes).toEqual(['*']);
```
Comment (vlasy) [inline src/test/acl.test.ts:199]: Why is this removed?

File: src/lib/AclPermission.ts [hunk 2]
```diff
@@ -21,15 +21,29 @@ export class AclPermission {
...
+        let result = this.attributes
+            .filter(attribute => attribute.charAt(0) !== '!')
             .map(attribute => jp.paths(data, `$..${attribute}`))
             .reduce((a, b) => a.concat(b), [])
-            .map((jpPath: string[]) => jpPath.slice(1))
-            .map((jpPath: string[]) => [jpPath, path(jpPath, data)])
-            .reduce((acc: any, [jpPath, data]: [string[], any]) => assocPath(jpPath, data, acc), {});
+            .map(jpPath => jpPath.slice(1))
+            .map(jpPath => [jpPath, path(jpPath, data)])
+            // @ts-ignore
```
Comment (vlasy) [inline src/lib/AclPermission.ts]: Please, specify ignored rule

**Review by vlasy — CHANGES_REQUESTED** [review]

File: src/test/acl.test.ts (hunk 1 above)
Comment (stefanprokopdev) [reply src/test/acl.test.ts:199]: I added `expect(permission.attributes.sort()).toEqual(['*', 'author', 'title']);`

The part `attributes: !attributes.includes('*') ? attributes : ['*'],` was changed. This is why `attributes` contains `author` and `title`. There should be changed the attributes filter - proposal. Should I create the issue?

File: src/lib/AclPermission.ts (hunk 2 above)
Comment (stefanprokopdev) [reply src/lib/AclPermission.ts]: The `jpPath` is `unknown` - I added the `as string[]`

File: src/test/acl.test.ts (hunk 1 above)
Comment (vlasy) [reply src/test/acl.test.ts:199]: Please, do.

**Review by vlasy — APPROVED** [review]

File: src/test/acl.test.ts (hunk 1 above)
Comment (stefanprokopdev) [reply src/test/acl.test.ts:199]: :heavy_check_mark: issue #8 created
